services/garden/gardenSlice.ts

```diff
@@ -43,6 +64,18 @@ export type GardenSlice = {
 
 export const gardenSelectors = {
   selectGardens: (state: RootState) => state.gardens,
-  // selectBed: (state: RootState, bedId: string) =>
-  //   state.garden.beds.find((bed) => bed.id === bedId),
+  selectBed: (state: RootState, gardenId: string, bedId: string) => {
+    // Coded immutably as Redux TK only wraps immer on reducers
+    const bed = state.gardens
+      .find((garden) => garden.id === gardenId)
+      ?.beds?.find((bed) => bed.id === bedId);
+
+    if (!bed?.veggies) return bed;
+
+    const veggiesWithInfo = bed.veggies.map((veggie) =>
+      appendVeggieInfoToVeggie(state, veggie)
+    );
+
+    return { ...bed, veggies: veggiesWithInfo };
```

Comment:
I've moved veggie.name to veggie.veggieInfo.name and generate name and image each time within the selector form veggieInfo.id. 
Then if we make any changes to veggieInfos it will keep all veggies updated too. Although depending on what we do with images and if we don't add any more info that's displayed with the veggie then it might be overkill

Reply:
I think we will need to add more info so definitely a good idea.